### PR TITLE
Change contained ValueSet reference from initial to extension

### PR DESCRIPTION
**Pull request by yunwwang**
Move the contained valueset reference from initial element to extension element.

### CODE REVIEW

File: server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java [hunk 1]
```diff
@@ -70,28 +74,19 @@ private void findAndReplaceValueSetReferences(List<QuestionnaireItemComponent> i
...
 
         // See if we need to add a reference or if the workaround was already done in the input
-        boolean initialItemNeeded = true;
-        for (QuestionnaireItemInitialComponent initialComponent : itemComponent.getInitial()) {
-          if (initialComponent.hasValueReference() 
-            && initialComponent.getValueReference().getReference().equals(itemComponent.getAnswerValueSet())) {
-              initialItemNeeded = false;
-            }
+        if (extensionList.stream().noneMatch(ext -> ext.equalsDeep(newExt))) {
+          extensionList.add(newExt);
```
[the remark this answers is not in the document]
Comment (yunwwang) [reply server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java]: fixed

File: server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java [hunk 2]
```diff
@@ -70,28 +74,19 @@ private void findAndReplaceValueSetReferences(List<QuestionnaireItemComponent> i
         // START WORKAROUNDS for HAPI Issue: https://github.com/jamesagnew/hapi-fhir/issues/1184
         // Add initial reference to any item with an answerValueSet due to HAPI encoder error.
```
[the remark this answers is not in the document]
Comment (yunwwang) [reply server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java]: fixed

File: server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java [hunk 3]
```diff
@@ -44,12 +45,15 @@ protected Questionnaire processResource(Questionnaire inputResource, FileStore f
    * Recursively visits every questionnaire item and replaces every `answerValueSet` that isn't a local
    * hash (#) reference into a local reference. This fills the valueSetMap with the loaded valuesets.
    * 
+   * @param questionnaire The questionnaire itself
```
[the remark this answers is not in the document]
Comment (yunwwang) [reply server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java]: parameter removed

File: server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java [hunk 4]
```diff
@@ -30,11 +30,18 @@ protected Questionnaire processResource(Questionnaire inputResource, FileStore f
...
     // Iterate through items recursively and replace answerValueSet appropiately
-    findAndReplaceValueSetReferences(inputResource.getItem(), inputResource.getExtension(), valueSetMap, fileStore, baseUrl);
+    findAndReplaceValueSetReferences(inputResource.getItem(), valueSetMap, fileStore, baseUrl);
 
     // Add all loaded valuesets to the contains field
     for (ValueSet valueSet : valueSetMap.values()) {
       inputResource.addContained(valueSet);
+
+      // START WORKAROUNDS for HAPI Issue: https://github.com/jamesagnew/hapi-fhir/issues/1184
+      // Add initial reference to any item with an answerValueSet due to HAPI encoder error.
```
Comment (hossenlopp) [inline server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java:40]: Update this comment to reflect what the new workaround is.

```suggestion
      // Add extension that references the contained ValueSet due to HAPI encoder error.
```

**Pull request merged**